### ISSUE TITLE
new sections for "Copy content from another course" cannot be used directly

### ISSUE THREAD

**Issue by tholudwig** [gjbarnard/moodle-format_topcoll#89]
Copying content from another course based in a section not used in the new course, leads to not shown content using topcall as course format. Being able to use the new section, you manually have to add the sections.

To reproduce:
1) Create a course1 with 4 sections.
2) Add one activity1 (label) to section4 in course1.
3) Create an new course (course2) using format_topcoll with just 1 section.
4) Copy activity1 from course1 to course 2.

Result: activity is not shown until you manually added 3 more sections to course2.

Maybe you can have a look a this. Ideally the topcoll is checking during import if new sections must be added. Alternative would be adding the activity to the last section section used in topcoll.

![topcoll_copy_activity](https://user-images.githubusercontent.com/14368866/99789965-b4142180-2b23-11eb-8dbf-76e218499fc3.png)
![topcoll_adding_sections](https://user-images.githubusercontent.com/14368866/99789967-b4acb800-2b23-11eb-8731-2433b916e700.png)

**Comment by gjb2048** [comment]
@tholudwig What version of Moodle and CT are you using please?  How do you accomplish step 4? MBZ = how??  What exact steps.

**Comment by tholudwig** [comment]
Hi @gjb2048 ,

currently we are using moodle 3.9.2 and topcoll 2019111702 (3.8). Maybe an update would help? That's something I have to check.

For Step 4.
4.1: Go to import in course administration in course1
4.2 Select course2 as source data to copy from
4.3 Just include activities and resources
4.4 Choose the section4 and activity1 to copy
4.5 Perform import

Best regards

**Comment by gjb2048** [comment]
@tholudwig Thank you for the information.  Please upgrade to be sure, but I don't think that will solve it, however there are always subtle differences.  May I ask if you think anybody reads this https://github.com/gjb2048/moodle-format_topcoll/tree/MOODLE_39#required-version-of-moodle please? - Its in the readme file too!

**Comment by tholudwig** [comment]
Unfortunately an update did not solve the problem. 

And yes we do read these instructions, but running a big Moodle with lots of plugins the naming of all the branches in these plugins can be quite confusing. With you it's crystal clear, but others just use release or master. But that's another story.

While preparing our own update it's quite possible, that we are ahead of some plugins. Maybe this is what happened here, because we directly switched from v3.5 to 3.9. But I'm not sure.

**Comment by gjb2048** [comment]
For step 4, don't you mean:

4.1 Go to import in course administration in course2
4.2 Select course1 as source data to copy from
4.3 Just include activities and resources
4.4 Choose the section4 and activity1 to copy
4.5 Perform import

??

**Comment by gjb2048** [comment]
Ok, done a few tests and while I can replicate, the class 'restore_format_topcoll_plugin' is not used by the 'import_ui' as it is with the 'restore_ui', so nothing I can do in code about this.  Additionally, no 'import_format_plugin' functionality, therefore please take this up on the Moodle Tracker that 'import' does not ask course formats if there is anything special it needs to do.

**Comment by John-Joubert** [comment]
Just a comment - I think this is normal functionality with the default moodle course formats.   Specifically, I've seen this when the default number of sections is lower than the number of sections imported.  For instance, if my default number of sections for a course is 5 sections, and I create a new course.  Then import a course with 10 sections - the last 5 sections are hidden, and I must edit the course to show the last 5 sections.  So - the question is - what is the default number of sections for the courses?  Is it 1?  Alternatively - after the destination course was created, was the number of visible sections lowered to 1?  By that I mean - was the number of sections visible less than the number of new sections that were imported? If so, this has nothing to do with Topcolll format, and this is just normal moodle behavior.

**Comment by John-Joubert** [comment]
Just a comment - I think this is normal functionality with the default moodle course formats. Specifically, I've seen this when the default number of sections is lower than the number of sections imported. For instance, if my default number of sections for a course is 5 sections, and I create a new course. Then import a course with 10 sections - the last 5 sections are hidden, and I must edit the course and increase the number of sections to show the last 5 sections. So - the question is - what is the default number of sections for the courses? Is it 1? Alternatively - after the destination course was created, was the number of visible sections lowered to 1? By that I mean - was the number of sections visible less than the number of new sections that were imported? If so, this has nothing to do with Topcolll format, and this is just normal moodle behavior.

Update: I just went and checked this very thing on my own moodle system. And I've confirmed this is normal behavior for standard moodle course formats (I tested it with "Topics Format"). So this isn't an issue with Collapsed Topics. This is just moodle.
